### ISSUE TITLE
Hash function can be optimized (more)

### ISSUE THREAD

**Issue by taisel** [threerings/clyde#15]
Sifting through hashCode implementations within clyde, there's excess arithmetic in some cases, and a lack of allowing for some collisions to occur within the bucketing, caused by excess precision in the hashCode functions. Some of the hash codes should collide if it means the O(1) time is lowered overall. Hashes that collide though should have a limit to the collision count in each bucket, causing a compare, if optimized.

tl;dr the hashCode implementations might need a thorough pass.

**Comment by taisel** [comment]
It's a dark art with performance tuning hashes, so I don't blame anyone for letting it sit like this.

**Comment by taisel** [comment]
In addition, I'm noticing unnecessary re-lookups.  Needs specific caching of locality of nodes outside of camera range in advance, to not traverse the entire scene every time.
